applications/software-libraries.md

```diff
@@ -0,0 +1,59 @@
+# Software Libraries
+
+The following describes the approach we adopt when making use of external software libraries in our applications.
+
+## Centrally capture your application's dependencies *in code*
+
+For Ruby this will primarily be in your **Gemfile** or equivalent, for Node in your **package.json** file, for Java in your **POM** file.
+
+This may seem like stating the obvious, but it is essential to allow the application to be checked for any issues in libraries that it depends on.
+
+## Capture your client-side dependencies too
+
+Ensure that any client-side libraries you use are captured in your dependencies.
+
+Again, this sounds obvious but not all client-side frameworks and libraries are necessarily included in your application through bundler or npm.
+
+> As an example: The GOV.UK frontend toolkit includes the GPL-licensed Nomensa accessible media player.  If you link to this in your client-side Javascript you will have to GPL *all* your client-side code.  Does the toolkit's MIT licence cover this?  No.  Is this dependency listed anywhere in the toolkit?  No.  Mind you, IANAL so maybe none of this matters.
+
+## Use a dependency checker
+
+With all your dependencies centrally listed, an automated dependency checker can be used to manage your exposure to
+
+- security vulnerabilities
+- bugs
+- licensing issues
+
+in your dependencies and all of their dependencies.
+
+For Rails projects we use Brakeman and [TBC...]
+
+For Node projects we use [TBC...]
+
+For Java projects we use [TBC...]
+
+## Check licences
+
+Before adding any library to your application, check the licence and copyright that applies to it.
+
+Any commercial licensing will need to be handled by the Commercial team.  Contact Ben Sagar in the first instance, who will be probably know who to start the conversation with in commercial.
+
+Any [copyleft](https://www.gnu.org/licenses/copyleft.en.html) licence conditions actually make it difficult for us to make our code as freely available as we would like so these should be avoided.  Any use of a library with such a licensing condition would need legal consideration.
+
+Generally speaking, if a library uses the [GNU GPL](https://www.gnu.org/licenses/licenses.en.html#GPL) then just don't use it.
+
```

Comment:
Suggest we also add comments about other common licenses, rather than just listing a couple of problematic ones.  In particular, MIT, BSD, Apache, LGPL.